### ISSUE TITLE
Isn't working with alpha,beta & rc up to final versions

### ISSUE THREAD

**Issue by nurb2kea** [iaian7/AN7-BlenderCheckForUpdates#1]
Hi,

I just checked with 3.6rc and 3.61rc.
Both say I'm on the latest (when the addon looks for patches or all updates)
With 3.6rc it should tell that there is the final 3.6 as update & tell me there is another daily update for it.

Means you won't get the daily updates and don't get the alpha, beta & rc to final updates.

Is it meant to only show non experimental updates like final 3.5 and 3.51 !?

**Comment by iaian7** [comment]
Hey @nurb2kea, thanks for posting the results from several release candidates!

The add-on can only check for official releases (https://download.blender.org/release/), and because it’s limited to version comparisons, it won’t detect alpha/beta/candidate nomenclature. 3.6.0 is still the latest official release, so 3.6.0rc and 3.6.1rc would both be detected as up-to-date based on their numbering. Blender stores versions as tuples; I suspect the Python tuple comparison is either ignoring the trailing characters or treating them as higher-value numbers, and of course 3.6.1 is newer than the available 3.6.0 release regardless.

Is there a plain text directory listing of prerelease version downloads similar to the final releases location I linked above? I haven’t come across anything like that yet, but let me know if it exists.

To answer your final question, yes; the plugin is only intended to notify users of final releases. And without a reliable way of finding pre-releases via automation (a plain text list of downloads with logical numbering), I don’t think it’s possible without significant development effort. The goal was to support artists working in LTS releases, and trying to build a tracking system for daily builds is well outside the intended scope of the project. But perhaps I may be able to implement final release notifications for users of alpha/beta/candidate versions? That might be doable, and I’ll keep it in mind as a potential feature in the next update.

**Comment by nurb2kea** [comment]
No it was just my thinking for my workflow.
I'm building and downloading the latest builds from alpha to final on a daily basis, to find bugs early to report and fix.
So sometimes it's hard to find all the different daily builds and their updates. 
So I thought I can save some time to get it checked and guided to the download of my version I'm running.

No need to change your addnon for just me, I was just curious!
Thanks for the answer :-)

**Comment by iaian7** [comment]
First of all, I’m glad you’re testing daily builds, thank you! Software testing is a lot of work.

Secondly, I can definitely see how it’d be a lot easier with an automated system with a download link. Unfortunately without a listing of available build downloads that’s reasonably easy to traverse as plain text with consistent naming schemes (the release directories are thankfully _mostly_ usable), that’s a little difficult. I’d like to avoid having to load the actual Blender website, especially since any HTML change could break the data processing. So if you ever find an rss feed or plain text directory listing of builds…let me know!

**Comment by nurb2kea** [comment]
Ok, I will !
Just an idea came to my mind. There is this blender launcher ! Maybe ask him how he get's all the details.
https://blenderartists.org/t/blender-launcher-standalone-software-client/1230543
![Screenshot 2023-07-16 at 1 44 48 am](https://github.com/iaian7/AN7-BlenderCheckForUpdates/assets/57676085/e186dcaa-f4d6-404b-b963-95a5e5fea987)

**Comment by iaian7** [comment]
Good idea! Based on the code, it looks like Blender Launcher is scraping the public Blender.org download pages with the help of some Python modules for parsing HTML content (it also hard-coded incorrect LTS releases? 3.6 is the current LTS version, but 3.7 was manually specified and apparently not updated when Blender development plans changed).

https://github.com/DotBow/Blender-Launcher/blob/v1.15.0-stable/source/threads/scraper.py

I haven't looked into this further (perhaps there's a Python module already included in Blender that would do the trick), but at first glance it's a level of complexity (and the potential slowness of loading public-facing pages with images) I was hoping to avoid. 🤔
